sklearn/model_selection/_search.py

```diff
@@ -591,7 +615,7 @@ def n_features_in_(self):
 
     @property
     def classes_(self):
-        self._check_is_fitted("classes_")
+        available_if(_estimator_has("classes_"))
```

Comment:
Do we need `available_if` here? 

```suggestion
        _estimator_has("classes_")(self)
```

I'm guessing `available_if` and properties do not work nicely together?

Reply:
> I'm guessing available_if and properties do not work nicely together?

Yes they don't.